app/models/framework/definition/transpiler.rb

```diff
@@ -0,0 +1,71 @@
+class Framework
+  module Definition
+    class Transpiler
+      attr_reader :ast
+
+      def initialize(ast)
+        @ast = ast
+      end
+
+      def transpile
+        ast = @ast # method-local binding required for Class.new blocks
+
+        @klass ||= Class.new(Framework::Definition::Base) do
+          framework_name         ast[:framework_name]
+          framework_short_name   ast[:framework_short_name]
+          management_charge_rate ast.dig(:management_charge, :flat_rate)
+        end.tap do |klass|
+          klass.const_set('Invoice', invoice_fields_class)
+        end
+      end
+
+      def invoice_fields_class
+        ast = @ast
+        transpiler = self
+
+        Class.new(Framework::EntryData) do
+          define_singleton_method :model_name do
```

Comment:
Yes. We weren't using the validations until the rake task arrived, at which case it blew up when validations were triggered because the class is anonymous and validations require classes to have names. This is the result of that, for which we'll add a comment to the commit.